python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py

```diff
@@ -52,26 +52,31 @@ def __init__(
         self._org_id = org_id
         self._log = log if log is not None else NullLogger()
 
-    async def generate_embeddings_async(self, texts: List[str]) -> ndarray:
+    async def generate_embeddings_async(
+        self, texts: List[str], batch_size: int = 0
+    ) -> ndarray:
         model_args = {}
         if self._api_type in ["azure", "azure_ad"]:
             model_args["engine"] = self._model_id
         else:
             model_args["model"] = self._model_id
 
         try:
-            response: Any = await openai.Embedding.acreate(
-                **model_args,
-                api_key=self._api_key,
-                api_type=self._api_type,
-                api_base=self._endpoint,
-                api_version=self._api_version,
-                organization=self._org_id,
-                input=texts,
-            )
-
-            # make numpy arrays from the response
-            raw_embeddings = [array(x["embedding"]) for x in response["data"]]
+            raw_embeddings = []
+            batch_size = batch_size or len(texts)
```

Comment:
if len(texts) > batch_size, (say 2 and 1) the batch_size will result in 2 which could cause unexpected behavior. 

```suggestion
            batch_size = min(batch_size, len(texts)
```

Reply:
Talked offline, this isn't the case